src/components/DeprovisionButton/index.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from "react";
+import { IServiceInstance } from "../../shared/ServiceCatalog";
+
+interface IDeprovisionButtonProps {
+  instance: IServiceInstance;
+  deprovision: (instance: IServiceInstance) => Promise<{}>;
+}
+
+interface IDeprovisionButtonState {
+  error: string | undefined;
+  instance: IServiceInstance | undefined;
+  isDeprovisioning: boolean;
+}
+
+class DeprovisionButton extends React.Component<IDeprovisionButtonProps, IDeprovisionButtonState> {
+  public state: IDeprovisionButtonState = {
+    error: undefined,
+    instance: this.props.instance,
+    isDeprovisioning: false,
+  };
+
+  public handleDeprovision = async () => {
+    const { deprovision, instance } = this.props;
+    this.setState({ isDeprovisioning: true });
+
+    try {
+      await deprovision(instance).then(async () => this.setState({ isDeprovisioning: false }));
```

Comment:
would it make sense to use the redux state to track this deprovisioning state? @evanlouie @Angelmmiguel?

Also, this might be more readable as:
```
await deprovision(instance)
this.setState({isDeprovisioning: false})
```